src/S3/MultipartUploader.php

```diff
@@ -134,10 +152,22 @@ protected function createPart($seekable, $number)
         }
 
         $data['ContentLength'] = $contentLength;
+        $this->uploadedBytes += $contentLength;
+        $this->displayProgress();
 
         return $data;
     }
 
+    protected function displayProgress()
+    {
+        $threshold = $this->progressThresholds;
+
+        if (!empty($threshold) and !empty($this->progressBar) and $this->uploadedBytes >= $threshold[0]) {
```

Comment:
'and' is a lot more common in python; we use && in this SDK.  This should also be split into three lines

Reply:
This also needs to be a while loop; what happens when we pass two thresholds at once (IE 4 part uploads)